web/src/osm2streets-svelte/Layer.svelte

```diff
@@ -0,0 +1,158 @@
+<script lang="ts">
+  import type { Feature, GeoJSON } from "geojson";
+  import type {
+    GeoJSONSource,
+    MapLayerMouseEvent,
+    MapMouseEvent,
+  } from "maplibre-gl";
+  import { onDestroy, onMount } from "svelte";
+  import { map } from "./store";
+  import { emptyGeojson, getLayerZorder } from "./utils";
+
+  // This component manages a GeoJSON source with exactly one MapLibre layer attached to it.
```

Comment:
A little bit of complexity here. Inspired by https://github.com/dimfeld/svelte-maplibre/, and one of the possible next steps is to refactor and just use that instead